otx/algorithms/classification/tasks/nncf.py

```diff
@@ -14,14 +14,50 @@
 # See the License for the specific language governing permissions
 # and limitations under the License.
 
-from otx.algorithms.classification.adapters.deep_object_reid.tasks import (
-    ClassificationNNCFTask,
-)
+from functools import partial
+from typing import Optional
+
+from mpa.utils.logger import get_logger
+
+from otx.algorithms.classification.adapters.mmcls.nncf import build_nncf_model
+from otx.algorithms.common.tasks.nncf_base import NNCFBaseTask
+from otx.api.entities.datasets import DatasetEntity
+from otx.api.entities.optimization_parameters import OptimizationParameters
 from otx.api.entities.task_environment import TaskEnvironment
+from otx.api.utils.argument_checks import check_input_parameters_type
+
+from .inference import ClassificationInferenceTask
+
 
+logger = get_logger()
 
-class OTXClassificationNNCFTask(ClassificationNNCFTask):
-    """Task for compressing classification models using NNCF."""
 
-    def __init__(self, task_environment: TaskEnvironment):  # pylint: disable=useless-parent-delegation
+class ClassificationNNCFTask(NNCFBaseTask, ClassificationInferenceTask):
+    @check_input_parameters_type()
+    def __init__(self, task_environment: TaskEnvironment):
         super().__init__(task_environment)
+        self._label_dictionary = dict(enumerate(self._labels, 1))
```

Comment:
Just curious, why start from 1?

Reply:
Thanks for catching it. This attribute is only for segmentation task. It is considered a bug and it would be removed in mmdeploy PR as this PR has been merged.